ports/esp32s2/README.md

```diff
@@ -0,0 +1,88 @@
+# Circuitpython on ESP32-S2 #
+
+This port adds the ESP32-S2 line of modules from Espressif to Circuitpython. ESP32-S2 modules are low power, single-core Wi-Fi microcontroller SoCs designed for IoT applications.
+
+## How this port is organized: ##
+
+- **bindings/** contains some required bindings to the ESP-IDF for exceptions and memory.
+- **boards/** contains the configuration files for each development board and breakout available on the port.
+- **common-hal/** contains the port-specific module implementations, used by shared-module and shared-bindings.
+- **esp-idf/** contains the Espressif IoT development framework installation, includign all the drivers for the port.
+- **modules/** contains information specific to certain ESP32-S2 hardware modules, such as the pins used for flash and RAM on the WROVER and WROOM.
+- **peripherals/** contains peripheral setup files and peripheral mapping information, sorted by family and sub-variant. Most files in this directory can be generated with the python scripts in **tools/**.
+- **supervisor/** contains port-specific implementations of internal flash, serial and USB, as well as the **port.c** file, which initializes the port at startup.
+- **tools/** includes useful python scripts for debugging and other purposes.
+
+At the root level, refer to **mpconfigboard.h** and **mpconfigport.mk** for port specific settings and a list of enabled circuitpython modules.
+
+## Connecting to the ESP32-S2 ##
+
+The USB port built into ESP32-S2 boards such as the Saola is not the native USB of the board, but a debugging and programming interface. The actual ESP32-S2 native USB which exposes the Circuitpython drive and CDC connection is located on IO pins 19 and 20:
+
+| GPIO | USB         |
+| ---- | ----------- |
+| 20   | D+ (green)  |
+| 19   | D- (white)  |
+| GND  | GND (black) |
+| 5V   | +5V (red)   |
+
+Connect these pins using a [USB adapter](https://www.adafruit.com/product/4090) or [breakout cable](https://www.adafruit.com/product/4448) to access the Circuitpython drive.
+
+## Building and flashing ##
+
+Before building or flashing the ESP32-S2, you must [install the esp-idf](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/get-started/index.html). This must be re-done ever time the esp-idf is updated, but not every time you build. From the port root, run:
+
+```
+./esp-idf/install.sh
+```
+
+Additionally, any time you open a new bash environment for building or flashing, you must add the esp-idf tools to your path:
+
+```
+. esp-idf/export.sh
+```
+
+To build for a board such as the Saola 1 Wrover and flash it over the UART interface, run the following:
+
+```
+make BOARD=espressif_saola_1_wrover flash
+```
```

Comment:
The MAkeFile default is set for MACOS https://github.com/adafruit/circuitpython/blob/main/ports/esp32s2/Makefile#L35
using PORT=/dev/tty.SLAB_USBtoUART
which works fine without being specified on my Mac

Foe Linux, I have to specify PORT=/dev/ttyUSB0  or whatever is correct

Reply:
Ok, I'll make this more vague, since it's the opposite for me.

Reply:
I don't understand how that can be? Same Makefile? Are you on Linux?

Reply:
I mean that on Mac, my ESP32-S2 boards do not appear as `tty.SLAB_USBtoUART`, but rather the `usbserial` name listed in my example. I thought I didn't have to do it on linux but it was a while ago that I tried so I'm probably misremembering. In any case, I'll just note that you need to specify the port.

Reply:
Ah -- OK -- so the Mac Section is fine -- might want to mention the SLAB_USBtoUARt is the default if it works....  but for Linux, the PORT=/dev/ttyUSBx  does have to be set.

Reply:
FWIW, I never saw the tty.SLAB_USBtoUART until UnexpectedMaker had me install some drivers.

I now find on MacOS I can connect to the the tty.usb* devices just fine.
2020-09-13 I built main and flashed with the following commands on MacOS. 

```
make BOARD=espressif_kaluga_1 PORT=/dev/tty.usbserial-1413201 flash    <-- over the external USB
make BOARD=espressif_saola_1_wrover PORT=/dev/tty.usbmodem01 flash     <-- over the esp32s2 native USB
make BOARD=unexpectedmaker_feathers2 PORT=/dev/tty.usbmodem01 flash    <-- over the esp32s2 native USB
```

Reply:
Very minor point...  tty.SLAB_USBtoUART* will have a number appended if there is more more than one USB-to-Serial device and the target of interest isn't the oldest one connected, so it may not work by default on macOS.